torch/distributed/distributed_c10d.py

```diff
@@ -686,7 +695,6 @@ def _new_process_group_helper(
 
     if not group_name:
         group_name = str(_group_count)
-        _group_count += 1
```

Comment:
We should keep all usage of `_group_count` together and at the beginning of this function to ensure consistency and correct usage (it's going to be easy to miss an increment when another early exit is added).

Reply:
Reverted `_group_count` changes and used `group_name` as group identifier for 3rd party backends.